src/main/webapp/script.js

```diff
@@ -0,0 +1,227 @@
+  const fileType = await getImageTypeOrError(file);
+
+  // check file size.
+  validateImageSize(file, fileType);
+
+  // File height or width can not be more than 1920px.
+  await validateImageDimensions(file);
+}
+
+/**
+ * Function to make sure the image is not too big.
+ * @param {File} image
+ * @param {string} imageType Can be 'png' or 'jpeg' only.
+ */
+function validateImageSize(image, imageType) {
+  if (imageType === 'png' && image.size > 8 * 1024 * 1024) {
+    throw new Error('File size should not exceed 8MB for png images.' +
+        'The size of an uploaded png image is ' +
+        Math.ceil(image.size / 1024 / 1024) + 'MB');
+  }
+
+  if (imageType === 'jpeg' && image.size > 2 * 1024 * 1024) {
+    throw new Error('File size should not exceed 2MB for jpeg images.' +
+        'The size of an uploaded jpeg image is ' +
+        Math.ceil(image.size / 1024 / 1024) + 'MB');
+  }
+}
+
+/**
+ * Function to get file type if it is png or jpeg
+ * and throw error otherwise
+ * @param {File} file
+ * @return {Promise}
+ */
+function getImageTypeOrError(file) {
+  return new Promise(function(resolve, reject) {
+    const fileReader = new FileReader();
+
+    fileReader.onloadend = function(e) {
+      // get first 4 bytes from file - they contain information
+      // about the extension.
+      const bytes = new Uint8Array(e.target.result);
+
+      // convert those bytes to string
+      let header = '';
+      for (let i = 0; i < bytes.length; i++) {
+        header += bytes[i].toString(16);
+      }
+
+      const PNG_HEADERS = ['89504e47'];
+      if (PNG_HEADERS.includes(header)) {
+        resolve('png');
+      }
+
+      const JPEG_HEADERS = ['ffd8ffe0', 'ffd8ffe1',
+        'ffd8ffe2', 'ffd8ffe3', 'ffd8ffe8'];
+      if (JPEG_HEADERS.includes(header)) {
+        resolve('jpeg');
+      }
+
+      reject(new Error(
+          'Invalid file type. Only jpeg and png images can be uploaded'));
+    };
+
+    // read first 4 bytes from file
+    fileReader.readAsArrayBuffer(file.slice(0, 4));
+  });
+}
+
+/**
+ * Function to make sure the image dimensions are no
+ * more than 1920 x 1080
+ * Promise returns error if this is not true and
+ * nothing otherwise.
+ * @param {File} imageFile
+ * @return {Promise}
+ */
+function validateImageDimensions(imageFile) {
+  return new Promise(function(resolve, reject) {
+    // construct image object from the file
+    const imageObject = new Image();
+    const imageUrl = URL.createObjectURL(imageFile);
+
+    imageObject.onload = function() {
+      if (imageObject.width > 1920) {
+        reject(new Error(
+            'The width of an image can not exceed 1920px. The width of ' +
+            'an uploaded image is' + imageObject.width + 'px'));
+        return;
+      }
+
+      if (imageObject.height > 1080) {
+        reject(new Error(
+            'The height of an image can not exceed 1080px. The height of ' +
+            'an uploaded image is' + imageObject.height + 'px'));
```

Comment:
I think there should be spaces surrounding the height variable, like `'an uploaded image is ' + imageObject.height + ' px'));`. Same for the width.